test/e2e/testsuites/testsuites.go

```diff
@@ -504,6 +528,149 @@ func (t *TestDeployment) Logs() ([]byte, error) {
+						Containers: []v1.Container{
+							{
+								Name:    "volume-tester",
+								Image:   imageutils.GetE2EImage(imageutils.BusyBox),
+								Command: []string{"/bin/sh"},
+								Args:    []string{"-c", command},
+								VolumeMounts: []v1.VolumeMount{
+									{
+										Name:      volumeName,
+										MountPath: mountPath,
+										ReadOnly:  readOnly,
+									},
+								},
+							},
+						},
+						RestartPolicy: v1.RestartPolicyAlways,
+					},
+				},
+				VolumeClaimTemplates: volumeClainTest,
+			},
+		},
+	}
+
+	if isWindows {
+		testStatefulset.statefulset.Spec.Template.Spec.NodeSelector = map[string]string{
+			"kubernetes.io/os": "windows",
+		}
+		testStatefulset.statefulset.Spec.Template.Spec.Containers[0].Image = "e2eteam/busybox:1.29"
+		testStatefulset.statefulset.Spec.Template.Spec.Containers[0].Command = []string{"powershell.exe"}
+		testStatefulset.statefulset.Spec.Template.Spec.Containers[0].Args = []string{"-Command", command}
+	}
+
+	return testStatefulset
+}
+
+func (t *TestStatefulset) Create() {
+	var err error
+	t.statefulset, err = t.client.AppsV1().StatefulSets(t.namespace.Name).Create(context.TODO(), t.statefulset, metav1.CreateOptions{})
+	framework.ExpectNoError(err)
+	err = waitForStatefulSetComplete(t.client, t.namespace, t.statefulset)
+	framework.ExpectNoError(err)
+	selector, err := metav1.LabelSelectorAsSelector(t.statefulset.Spec.Selector)
+	framework.ExpectNoError(err)
+	options := metav1.ListOptions{LabelSelector: selector.String()}
+	statefulSetPods, err := t.client.CoreV1().Pods(t.namespace.Name).List(context.TODO(), options)
+	framework.ExpectNoError(err)
+	// always get first pod as there should only be one
+	t.podName = statefulSetPods.Items[0].Name
+}
+
+func (t *TestStatefulset) WaitForPodReady() {
+	selector, err := metav1.LabelSelectorAsSelector(t.statefulset.Spec.Selector)
+	framework.ExpectNoError(err)
+	options := metav1.ListOptions{LabelSelector: selector.String()}
+	statefulSetPods, err := t.client.CoreV1().Pods(t.namespace.Name).List(context.TODO(), options)
+	framework.ExpectNoError(err)
+	// always get first pod as there should only be one
+	pod := statefulSetPods.Items[0]
+	t.podName = pod.Name
+	err = e2epod.WaitForPodRunningInNamespace(t.client, &pod)
+	time.Sleep(40 * time.Second)
+	framework.ExpectNoError(err)
+}
+
+func (t *TestStatefulset) Exec(command []string, expectedString string) {
+	_, err := framework.LookForStringInPodExec(t.namespace.Name, t.podName, command, expectedString, execTimeout)
+	framework.ExpectNoError(err)
+}
+
+func (t *TestStatefulset) DeletePodAndWait() {
+	e2elog.Logf("Deleting pod %q in namespace %q", t.podName, t.namespace.Name)
+	err := t.client.CoreV1().Pods(t.namespace.Name).Delete(context.TODO(), t.podName, metav1.DeleteOptions{})
+	if err != nil {
+		if !apierrs.IsNotFound(err) {
+			framework.ExpectNoError(fmt.Errorf("pod %q Delete API error: %v", t.podName, err))
+		}
+		return
+	}
+	time.Sleep(40 * time.Second)
```

Comment:
why sleep here?

Reply:
After pod is deleted, waitForPodready will  ensure the new pod is ready. If don't sleep here,waitForPodready will be passed before old pod is deleted.

Reply:
ok, pls add a comment before `time.Sleep`.

Reply:
so statefulset is different from deployment, in deployment, it's not necessary to wait

Reply:
Actually,in deployment also use wait,but it use funcition like this.
```
err = e2epod.WaitForPodNoLongerRunningInNamespace(t.client, t.podName, t.namespace.Name)
```
But in statefulset situation, If I use this function,the deleted pod will never be NoLongerRunning State as the new pod has the same name and namespace with old one. So after the new pod created,the test will timeout here. And I choose to use sleep to wait.